docs/src/components/ServiceTest.tsx

```diff
@@ -92,65 +50,69 @@ export default function ServiceTest(props: { serviceSpec: jdspec.ServiceSpec })
+                                    <Card key={service.id}>
+                                        <DeviceCardHeader device={service.device} />
+                                        <CardActions>
+                                            <Button variant="contained" color="primary" onClick={handleSelect(service)}>Select</Button>
+                                        </CardActions>
+                                    </Card>
+                                </Grid>)}
+                            </Grid>}
+                            {!serviceInstances.length && <Alert severity="info">Not seeing your device? Try some of the following.
+                        <ul>
+                                    <li>Check that your device is connected</li>
+                                    <li>Use the <strong>packet console</strong> to monitor packets on the bus</li>
+                                    <li>Check the class identifier in your annoucement packets</li>
+                                </ul></Alert>}
                         </StepContent>
                     </Step>
-                ))}
-            </Stepper>
-            {activeStep === stepLength && (
-                <Paper square elevation={0} className={classes.resetContainer}>
-                    <Typography>All steps completed - you&apos;re finished</Typography>
-                    <Button onClick={handleReset} className={classes.button}>
-                        Reset
-          </Button>
-                </Paper>
-            )}
-        </div>
+                    {serviceTest.tests.map((test, index) => (
+                        <Step key={index}>
+                            <StepLabel>{test.description}</StepLabel>
+                            <StepContent>
+                                {index != activeTest && <Button onClick={handleStartTest(index)} >
+                                    Start Test
+                                </Button>}
+                                {index === activeTest && <ServiceUnitTest
+                                    serviceInstance={selectedServiceInstance}
+                                    test={test}
+                                    finished={testFinished}
+                                >
+                                </ServiceUnitTest>}
+                            </StepContent>
+                        </Step>
+                    ))}
+                </Stepper>
+                {activeStep === serviceTest.tests.length + 1 && (
+                    <Paper square elevation={0} >
+                        <Typography>All steps completed - you're finished</Typography>
+                        <Button onClick={handleReset} >
+                            Reset
+                </Button>
+                    </Paper>
+                )}
+            </Grid>
+            {selectedServiceInstance && <DashbardDeviceItem
+                key={selectedServiceInstance.device.id}
```

Comment:
not needed, only for collections

Reply:
We need to show the device twin so when the user steps through the test, they see changes.

Reply:
oh